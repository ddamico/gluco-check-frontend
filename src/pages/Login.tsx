import React from "react";
import * as firebase from "firebase/app";
import * as firebaseui from "firebaseui";
import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
import { auth } from "../lib/firebase";
import { useTranslation } from "react-i18next";

function Login() {
  const firebaseUIConfig = {
    credentialHelper: firebaseui.auth.CredentialHelper.NONE,
    signInFlow: "redirect",
    signInSuccessUrl: "/",
    signInOptions: [
      // firebase.auth.EmailAuthProvider.PROVIDER_ID,
      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
      // firebase.auth.FacebookAuthProvider.PROVIDER_ID,
      // firebase.auth.TwitterAuthProvider.PROVIDER_ID,
      // firebase.auth.GithubAuthProvider.PROVIDER_ID,
      // "apple.com",
    ],
  };

  const { t } = useTranslation();

  return (
    <div>
      <h2>{t("login.title")}</h2>
      <StyledFirebaseAuth uiConfig={firebaseUIConfig} firebaseAuth={auth} />
    </div>
  );
}

export default Login;
