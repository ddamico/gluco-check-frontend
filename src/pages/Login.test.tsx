import React from "react";
import { cleanup, render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import Login from "./Login";

expect.extend(toHaveNoViolations);

jest.mock("../lib/firebase");
jest.mock("react-firebaseui/StyledFirebaseAuth", () => {
  return {
    __esModule: true,
    default: () => {
      return <>This is where the login would be</>;
    },
  };
});

afterEach(cleanup);

describe("Login page", () => {
  it("renders the component", () => {
    const { container } = render(<Login />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("has no axe violations", async () => {
    const { container } = render(<Login />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
