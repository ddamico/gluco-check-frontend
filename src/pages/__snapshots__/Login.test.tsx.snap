// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders the component 1`] = `
<div>
  <h2>
    login.title
  </h2>
  This is where the login would be
</div>
`;
